Cython/Includes/cpython/datetime.pxd

```diff
@@ -89,9 +150,22 @@ cdef extern from "datetime.h":
         object (*Time_FromTime)(int, int, int, int, object, PyTypeObject*)
         object (*Delta_FromDelta)(int, int, int, int, PyTypeObject*)
 
+        # We cannot use the following because we still need to compile on previous versions,
+        # instead we use datetime.h's macros that we can do some C macro magic on top of.
+
+        # constructors
+        # object (*TimeZone_FromTimeZone)(PyObject *offset, PyObject *name)
+
+        # singletons
+        # PyObject *TimeZone_UTC
+
         # constructors for the DB API
-        object (*DateTime_FromTimestamp)(object, object, object)
-        object (*Date_FromTimestamp)(object, object)
+        # object (*DateTime_FromTimestamp)(PyTypeObject*, object, PyObject*)
+        # object (*Date_FromTimestamp)(PyTypeObject*, object)
+
+        # PEP 495 constructors
+        # object (*DateTime_FromDateAndTimeAndFold)(int, int, int, int, int, int, int, object, int, PyTypeObject*)
+        # object (*Time_FromTimeAndFold)(int, int, int ,int, object, int, PyTypeObject*)
```

Comment:
```suggestion
        object (*DateTime_FromTimestamp)(object, object, PyObject*)
        object (*Date_FromTimestamp)(object, object)

        # PEP 495 constructors
        object (*DateTime_FromDateAndTimeAndFold)(int, int, int, int, int, int, int, object, int, PyTypeObject*)
        object (*Time_FromTimeAndFold)(int, int, int ,int, object, int, PyTypeObject*)
```

Reply:
```diff
+        object (*DateTime_FromTimestamp)(object, object, PyObject*)
+        object (*Date_FromTimestamp)(object, object)
```

This doesn't work, see:
```
------------------------------------------------------------
...
cdef inline object timezone_new(int offset, str name=''):
    return PyTimeZone_FromOffsetAndName(offset, name)

# Create datetime object using DB API constructor.
cdef inline object datetime_from_timestamp(int timestamp, object tz=None):
    return PyDateTimeAPI.DateTime_FromTimestamp(PyDateTimeAPI.DateTimeType, (timestamp, tz) if tz is not None else (timestamp,), NULL)
                                                            ^
------------------------------------------------------------

datetime.pxd:265:61: Cannot convert 'PyTypeObject *' to Python object
```

Reply:
That should be solvable with a cast: `<object>PyDateTimeAPI.DateType`

Reply:
_Right_, but why cast it when you can just define it as a `PyTypeObject *` in the first place?